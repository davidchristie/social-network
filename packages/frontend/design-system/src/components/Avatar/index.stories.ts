import Avatar from ".";
import addComponentStory from "../../utilities/storybook/addComponentStory";

addComponentStory(Avatar, {
  children: undefined,
});
