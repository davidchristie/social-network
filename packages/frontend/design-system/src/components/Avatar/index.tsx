import MaterialAvatar from "@material-ui/core/Avatar";
import withStyles, {
  WithStyles
} from "@material-ui/core/styles/withStyles";
import React from "react";

import Icon from "../Icon";

interface Props {
  children?: React.ReactNode;
  image?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
  size?: Size;
}

const sizes = {
  large: {
    height: 60,
    width: 60,
  },
  small: {},
};

const styles = {
  ...sizes,
};

type Size = keyof typeof sizes;

const Content: React.SFC<Props & WithStyles<"large"> & WithStyles<"small">> = ({
  children,
  classes,
  image,
  size = "small",
  ...attributes
}) => {
  return (
    <MaterialAvatar
      className={classes[size]}
      src={image}
      {...attributes}
    >
      {children}
      {!children && !image && (
        <Icon size={size}>
          account_circle
        </Icon>
      )}
    </MaterialAvatar>
  );
};

const Styled = withStyles(styles)(Content);

const Avatar: React.SFC<Props> = props => <Styled {...props} />;

export default Avatar;
