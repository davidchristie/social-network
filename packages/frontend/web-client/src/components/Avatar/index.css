.Avatar {
  background: white;
  border: 1px solid black;
  border-radius: 50%;
}

.Avatar.small {
  height: 35px;
  width: 35px;
}

.Avatar.large {
  height: 70px;
  width: 70px;
}
