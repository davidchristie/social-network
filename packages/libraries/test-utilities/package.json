{
  "name": "test-utilities",
  "private": true,
  "scripts": {
    "build": "rimraf dist && tsc"
  },
  "version": "1.0.0"
}
