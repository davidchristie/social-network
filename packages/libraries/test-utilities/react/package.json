{
  "main": "../dist/react/index.js",
  "name": "test-utilities/react",
  "private": true,
  "version": "1.0.0"
}
