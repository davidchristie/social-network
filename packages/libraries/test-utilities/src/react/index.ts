export { default as describeMountedElement } from "./describeMountedElement";
export { default as describeWithCustomClass } from "./describeWithCustomClass";
export { default as describeWithNoProps } from "./describeWithNoProps";
export { default as describeWithProps } from "./describeWithProps";
export { default as itMatchesSnapshot } from "./itMatchesSnapshot";
