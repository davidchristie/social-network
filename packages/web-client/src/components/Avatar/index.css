.Avatar {
  border-radius: 50%;
}

.Avatar.small {
  height: 35px;
  width: 35px;
}

.Avatar.large {
  height: 70px;
  width: 70px;
}
