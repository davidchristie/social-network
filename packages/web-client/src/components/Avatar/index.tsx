import React from "react";

import "./index.css";

interface Props {
  image: string;
  onClick?: (event: React.MouseEvent<HTMLImageElement>) => void;
  size: "small" | "large";
}

export default class Avatar extends React.Component<Props> {
  public render () {
    return (
      <img
        className={`Avatar ${this.props.size}`}
        onClick={this.props.onClick}
        src={this.props.image}
      />
    );
  }
}
