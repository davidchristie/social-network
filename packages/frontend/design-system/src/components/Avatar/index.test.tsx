import React from "react";
import { describeMountedElement } from "test-utilities/react";

import Avatar from ".";

describe("Avatar component", () => {
  describeMountedElement("with no props", <Avatar />);

  describeMountedElement(`with size "small"`, <Avatar size="small" />);

  describeMountedElement(`with size "large"`, <Avatar size="large" />);
});
