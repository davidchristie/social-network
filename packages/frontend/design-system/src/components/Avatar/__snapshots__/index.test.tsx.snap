// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component with no props matches shapshot 1`] = `
<Avatar>
  <WithStyles(Content)>
    <Content
      classes={
        Object {
          "large": "Content-large-1",
          "small": "Content-small-2",
        }
      }
    >
      <WithStyles(Avatar)
        className="Content-small-2"
      >
        <Avatar
          className="Content-small-2"
          classes={
            Object {
              "colorDefault": "MuiAvatar-colorDefault-4",
              "img": "MuiAvatar-img-5",
              "root": "MuiAvatar-root-3",
            }
          }
          component="div"
        >
          <div
            className="MuiAvatar-root-3 MuiAvatar-colorDefault-4 Content-small-2"
          >
            <Icon
              size="small"
            >
              <WithStyles(Icon)
                color="inherit"
                fontSize="small"
              >
                <Icon
                  classes={
                    Object {
                      "colorAction": "MuiIcon-colorAction-9",
                      "colorDisabled": "MuiIcon-colorDisabled-11",
                      "colorError": "MuiIcon-colorError-10",
                      "colorPrimary": "MuiIcon-colorPrimary-7",
                      "colorSecondary": "MuiIcon-colorSecondary-8",
                      "fontSizeInherit": "MuiIcon-fontSizeInherit-12",
                      "fontSizeLarge": "MuiIcon-fontSizeLarge-14",
                      "fontSizeSmall": "MuiIcon-fontSizeSmall-13",
                      "root": "MuiIcon-root-6",
                    }
                  }
                  color="inherit"
                  component="span"
                  fontSize="small"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-6 MuiIcon-fontSizeSmall-13"
                  >
                    account_circle
                  </span>
                </Icon>
              </WithStyles(Icon)>
            </Icon>
          </div>
        </Avatar>
      </WithStyles(Avatar)>
    </Content>
  </WithStyles(Content)>
</Avatar>
`;

exports[`Avatar component with size "large" matches shapshot 1`] = `
<Avatar
  size="large"
>
  <WithStyles(Content)
    size="large"
  >
    <Content
      classes={
        Object {
          "large": "Content-large-1",
          "small": "Content-small-2",
        }
      }
      size="large"
    >
      <WithStyles(Avatar)
        className="Content-large-1"
      >
        <Avatar
          className="Content-large-1"
          classes={
            Object {
              "colorDefault": "MuiAvatar-colorDefault-4",
              "img": "MuiAvatar-img-5",
              "root": "MuiAvatar-root-3",
            }
          }
          component="div"
        >
          <div
            className="MuiAvatar-root-3 MuiAvatar-colorDefault-4 Content-large-1"
          >
            <Icon
              size="large"
            >
              <WithStyles(Icon)
                color="inherit"
                fontSize="large"
              >
                <Icon
                  classes={
                    Object {
                      "colorAction": "MuiIcon-colorAction-9",
                      "colorDisabled": "MuiIcon-colorDisabled-11",
                      "colorError": "MuiIcon-colorError-10",
                      "colorPrimary": "MuiIcon-colorPrimary-7",
                      "colorSecondary": "MuiIcon-colorSecondary-8",
                      "fontSizeInherit": "MuiIcon-fontSizeInherit-12",
                      "fontSizeLarge": "MuiIcon-fontSizeLarge-14",
                      "fontSizeSmall": "MuiIcon-fontSizeSmall-13",
                      "root": "MuiIcon-root-6",
                    }
                  }
                  color="inherit"
                  component="span"
                  fontSize="large"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-6 MuiIcon-fontSizeLarge-14"
                  >
                    account_circle
                  </span>
                </Icon>
              </WithStyles(Icon)>
            </Icon>
          </div>
        </Avatar>
      </WithStyles(Avatar)>
    </Content>
  </WithStyles(Content)>
</Avatar>
`;

exports[`Avatar component with size "small" matches shapshot 1`] = `
<Avatar
  size="small"
>
  <WithStyles(Content)
    size="small"
  >
    <Content
      classes={
        Object {
          "large": "Content-large-1",
          "small": "Content-small-2",
        }
      }
      size="small"
    >
      <WithStyles(Avatar)
        className="Content-small-2"
      >
        <Avatar
          className="Content-small-2"
          classes={
            Object {
              "colorDefault": "MuiAvatar-colorDefault-4",
              "img": "MuiAvatar-img-5",
              "root": "MuiAvatar-root-3",
            }
          }
          component="div"
        >
          <div
            className="MuiAvatar-root-3 MuiAvatar-colorDefault-4 Content-small-2"
          >
            <Icon
              size="small"
            >
              <WithStyles(Icon)
                color="inherit"
                fontSize="small"
              >
                <Icon
                  classes={
                    Object {
                      "colorAction": "MuiIcon-colorAction-9",
                      "colorDisabled": "MuiIcon-colorDisabled-11",
                      "colorError": "MuiIcon-colorError-10",
                      "colorPrimary": "MuiIcon-colorPrimary-7",
                      "colorSecondary": "MuiIcon-colorSecondary-8",
                      "fontSizeInherit": "MuiIcon-fontSizeInherit-12",
                      "fontSizeLarge": "MuiIcon-fontSizeLarge-14",
                      "fontSizeSmall": "MuiIcon-fontSizeSmall-13",
                      "root": "MuiIcon-root-6",
                    }
                  }
                  color="inherit"
                  component="span"
                  fontSize="small"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-6 MuiIcon-fontSizeSmall-13"
                  >
                    account_circle
                  </span>
                </Icon>
              </WithStyles(Icon)>
            </Icon>
          </div>
        </Avatar>
      </WithStyles(Avatar)>
    </Content>
  </WithStyles(Content)>
</Avatar>
`;
